models/repo.go

```diff
@@ -1275,6 +1275,20 @@ func updateRepository(e Engine, repo *Repository, visibilityChanged bool) (err e
 			}
 		}
 
+		// Create/Remove git-daemon-export-ok for git-daemon...
+		daemonExportFile := path.Join(repo.RepoPath(), `git-daemon-export-ok`)
+		if repo.IsPrivate {
+			if err = os.Remove(daemonExportFile); err != nil {
```

Comment:
It will return a `*PathError` with the message `File does not exist`, but I'd rather have false positives than not getting `file can't be removed because permissions`-errors :unamused: